modules/authentication/proc-creating-a-custom-transformer-to-provision-users-from-rhsso-to-the-software-catalog.adoc

```diff
@@ -0,0 +1,92 @@
+[id="provisioning-users-from-rhsso-to-the-software-catalog"]
+= Creating a custom transformer to provision users from Red Hat Single-Sign On (RHSSO) to the software catalog
+
+To customize further how users/groups are mapped into entities in {product}, you can create a custom backend module that utilizes the `keycloakTransformerExtensionPoint` to provide custom user and group transformers to the keycloak backend.
+
+.Prerequisites
+* You xref:provisioning-users-from-rhsso-to-the-software-catalog[enabled provisioning users from Red Hat Single-Sign On (RHSSO) to the software catalog].
+
+.Procedure
+. Create a new backend module with the `yarn new` command.
+
+. Add your custom user and group transformers to the `keycloakTransformerExtensionPoint`.
+
++
+Below is an example of how the backend module can be defined:
++
+.`plugins/__<module-name>__/src/module.ts`
+[source,javascript]
+----
+/* highlight-add-start */
+import {
+  GroupTransformer,
+  keycloakTransformerExtensionPoint,
+  UserTransformer,
+} from '@janus-idp/backstage-plugin-keycloak-backend';
+
+const customGroupTransformer: GroupTransformer = async (
+  entity,
+  realm,
+  groups,
+) => {
+  /* apply transformations */
+  return entity;
+};
+const customUserTransformer: UserTransformer = async (
+  entity,
+  user,
+  realm,
+  groups,
+) => {
+  /* apply transformations */
+  return entity;
+};
+/* highlight-add-end */
+
+export const keycloakBackendModuleTransformer = createBackendModule({
+  pluginId: 'catalog',
+  moduleId: 'keycloak-transformer',
+  register(reg) {
+    reg.registerInit({
+      deps: {
+        /* highlight-add-start */
+        keycloak: keycloakTransformerExtensionPoint,
+        /* highlight-add-end */
+      },
+      /* highlight-add-start */
+      async init({ keycloak }) {
+        keycloak.setUserTransformer(customUserTransformer);
+        keycloak.setGroupTransformer(customGroupTransformer);
+        /* highlight-add-end */
+      },
+    });
+  },
+});
+----
++
+[IMPORTANT]
+====
+The `pluginId` for the module **MUST** be set to `catalog` to match the `pluginId` of the `keycloak-backend` or else the module will fail to initialize.
+====
+
+. Install this new backend module into your {product-short} backend.
```

Comment:
May consider adding example on how to do this like: 
`backend.add(import(backstage-plugin-catalog-backend-module-keycloak-transformer))`

Reply:
Like this?

```suggestion
. Install this new backend module into your {product-short} backend.
+
[source,javascript]
----
backend.add(import(backstage-plugin-catalog-backend-module-keycloak-transformer))
----
```

Reply:
Yes that looks good!